### ISSUE TITLE
Broken launchPatient -- Bad NotificationContext Unique Constraint

### ISSUE THREAD

**Issue by claylong** [drajer-health/eCRNow#744]
A new uniqueness constraint was added to the NotificationContext class that was intended to be deployed to the DB.

Commit:  d58c7757dbe76f2c7d92e56a77cbca0ff5f2ec11
Change:  https://github.com/drajer-health/eCRNow/commit/d58c7757dbe76f2c7d92e56a77cbca0ff5f2ec11#diff-7ca39eefd6ca83cb0f4bc773c6c3c0b3d7de008b3caa77649f19cdf15aea3082

There are several issues with this change:

1. The @UniqueConstraint doesn't specify a name, so the ORM library generates a junk string to use as the name (ours was called "ukijmdj1gmqols0g57mp7wp8ijy")
2. This constraint cannot be applied to an existing DB if the data doesn't meet the uniqueness requirement, which was previously acceptable in the ECRNow code.  This results in the constraint missing in the DB after an ECRNow upgrade.  It is only present when ECRNow is deployed with a clean DB, which should not be a requirement of ECRNow upgrade.
3. The constraint doesn't seem to correct, at all, with the controller and business logic.  It was introduced in 3.1.2, at which time it would break the existing behavior for launchPatient.  There was a *[later commit](https://github.com/drajer-health/eCRNow/commit/193dd3f9082ffbda2ab9c7b1d4777b6cda894965)* that wasn't available until 3.1.4 that did try to use the constraint to disallow "[duplicate launches for launchPatient](https://github.com/drajer-health/eCRNow/releases/tag/v3.1.4_new)", but that functionality was technically already broken in 3.1.2. 

TL;DR -- The constraint was wrong, introduced too early, and couldn't be applied correctly.  Luckily this means ECRNow upgrades just *happen* to work (b/c the two bugs cancel each other out), but new ECRNow implementations behave **_differently_**, causing problems with existing functionality. 

This effectively breaks the existing behavior of launchPatient, w/o reLaunch being fixed/ready for use instead.  This breaks basic launch functionality.

**Comment by nbashyam** [comment]
This is a required constraint going forward to avoid overloading of EHRs with queries. 
Please provide any thoughts on how best to migrate existing databases so that we can provide upgrade scripts.

**Comment by mmatiasalt** [comment]
@claylong do you have any suggestions or additional comments for Nagesh, or may we close this ticket? Thank you.

**Comment by bhargavchowdhary** [comment]
We will create a migration script.
